### ISSUE TITLE
Problems with free text entry (Affects current pilot study) 

### ISSUE THREAD

**Issue by hotavocado** [ChildMindInstitute/mindlogger-app#1604]
Problem 1: The keyboard blocks the text box when a users goes to enter text, they cannot see what they are typing. This affects both android and iOS. See video for example: https://vimeo.com/518894860#t=58

Problem 2: The text box is only one line, and the text does not wrap. After the user finishes typing a long text response, it's difficult for the user to review what he/she has written. 

Proposed redesign to address both problems:

1. While entering text

a. A text box with a 'done' button on the right side will appear
b. The box is of fixed size, user can scroll within the box
c. Hitting return on the keyboard creates new line
d. Hitting the done button hides keyboard and exits text entry

| before | after |
| ------------- | ------------- |
| <img width="400" alt="Screen Shot 2021-03-04 at 2 43 51 PM" src="https://user-images.githubusercontent.com/24463939/110032203-deba6c80-7d05-11eb-8836-18d4ecdc7145.png"> | <img width="400" alt="mockup1" src="https://user-images.githubusercontent.com/24463939/110032343-06113980-7d06-11eb-9f3d-6019f4f03b20.png"> |

2. After entering text

a. A box that wraps text will appear, instead of the current single line display
b. The height of this box depends on the length of the response, the user can scroll down the screen to review the entire response
c. Tapping the box re-enters text edit

| before | after |
| ------------- | ------------- |
<img width="400" alt="Screen Shot 2021-03-04 at 3 02 57 PM" src="https://user-images.githubusercontent.com/24463939/110032739-7ddf6400-7d06-11eb-9a78-696e47e8a2f3.png"> | <img width="400" alt="mockup 2" src="https://user-images.githubusercontent.com/24463939/110032787-8d5ead00-7d06-11eb-8c4c-8315b8cdf70d.png"> |
 |         |  <img width="400" alt="mockup 2.1" src="https://user-images.githubusercontent.com/24463939/110035114-3dcdb080-7d09-11eb-9bca-3f48b4a619e5.png"> |

Who this affects:

Almost every applet created for the current clinical pilot study has free text responses, where the user leaves notes, comments, and observations.

**Comment by sergeysemenuta** [comment]
The only thing which is fixed is - The keyboard blocks the text box . This issue is not reproduced
https://www.screencast.com/t/3OoFbb0E0
ML v 0.15.17
Pixel 4XL//Android 11
iPhone X // iOS 13.3
Samsung Galaxy S8 //Android 8

**Comment by WorldImpex** [comment]
@devbtech Please revert this feature and replace it with the original version of a textbox. Make sure that version of the textbox is not blocked by the keyboard as well

**Comment by sergeysemenuta** [comment]
The feature is reverted to the original version. The textbox is not blocked by the keyboard

I discovered only one specific issue on Pixel 4 XL.
On this device, we have the bottom bar over the keyboard and it is a little overlapping the textbox after clicking on it. When the user starts typing the text is fully visible.
https://images.zenhubusercontent.com/103405139/5d9424eb-42e9-41c7-96af-c70488f99f28/2021_03_16_14_25_32.mp4

ML v 0.15.20
Pixel 4XL//Android 11
iPhone X // iOS 13.3
Samsung Galaxy S8 //Android 8
Samsung Galaxy S8 //Android 8

**Comment by sergeysemenuta** [comment]
- The text option is disabled on the applet builder. I'm not able to create an applet with input text box
- On the original text box item, I do not see any updates. Text is displayed in one single line 
https://admin-staging.mindlogger.org/
ML v 0.15.29

**Comment by WorldImpex** [comment]
Optional textbox is not updated

**Comment by sergeysemenuta** [comment]
The text option is enabled on the applet builder on the applet builder.

- The keyboard hides the textbox when a user is entering the text. this issue is reproducible for the textbox item

Expected result: 

1. Once the user taps into the textbox and start typing he/she should be able to see the text he is entering on
2. The user should be able to see all the text  when clicking the return button on the keyboard 

Another case (text input option is enabled for the radio checkbox slider etc. items)

- The entered text is displayed in one line

Expected: the text should be wrapped in the next line after tapping the return button on the keyboard
Video iphoneX:https://www.screencast.com/t/5eaudILiw
Video iPhone 7: https://www.screencast.com/t/zFuZu9HGY
Owner's account: snakessqa@gmail.com // 123456
Applet: #1604
Activity: #1604(contain the text items)
Activity #1604 text input option (contain the text input options for the rest of item types)
Applet password: Qwe123!!!


On android device Pixel 4XL//Android 11   works as expected for the text items but doesn't work for the text input option enabled
Another issue is that the 'next' button doesn't appear after entering text
The top part of the placeholder is cut off
**Video Android:** https://www.screencast.com/t/IcHazdQXiu
Environment:
https://admin-staging.mindlogger.org/
Pixel 4XL//Android 11
iPhone 7// iOS 13.4.1
iPhone X // iOS 13.3

**Comment by sergeysemenuta** [comment]
Note: The next button shows up on the text items when the user hides the keyboard manually and taps outside of the textbox

**Comment by sergeysemenuta** [comment]
On Pixel 4xl really big white space area which hides the text and images
See the video: https://www.screencast.com/t/1rhAqLFQt
On the input option, the font size is lower than on the text items text box. The same behavior on iOS
Expected: Font size should be the same
https://www.screencast.com/t/V6Sco6mvEjj2
On iPhone X different behavior of scrolling in text input option textbox and text item textbox
https://www.screencast.com/t/TvForWSr55
Staging
ML v 0.16.4
iPhone X // iOS 13.3
Pixel 4XL//Android 11

**Comment by WorldImpex** [comment]
@mtg137 Lets try to fix the google pixel issue and the other two we can backlog

**Comment by Matchdev120** [comment]
@mtg137 I can see you used KeyboardAwareScrollView for iOS devices to #1604 but it occurs 
So I'm changing it back to behavior={"padding"} on iOS.

**Comment by mtg137** [comment]
@Matchdev120 
Regarding this comment, I am already using KeyboardAwareScrollView.
`You should use KeyboardAwareScrollView behavior={"padding"} on iOS.`

what do you mean?

**Comment by sergeysemenuta** [comment]
@devbtech On which version of the app should I check?

**Comment by sergeysemenuta** [comment]
On ML v 0.16.5 I can't scroll and do not see the text
https://www.screencast.com/t/BUXjp6sNiS
 Staging
ML v 0.16.5
iPhone X // iOS 13.3

**Comment by Matchdev120** [comment]
@mtg137 I wanted to say this one:
KeyboardAvoidingView behavior={"padding"} on iOS

**Comment by sergeysemenuta** [comment]
- The textbox is not visible on android it's hidden. It is not possible to view the text and scroll it. It depends on how many radio buttons in one screen
- On the  ios devices on radio and checkboxes screens the text box is too small it is not convenient to scroll in order to view the text

**Video:** https://www.screencast.com/t/xhHsAXsg15
Checked on:
Staging
ML v 0.16.8
Pixel 4XL//Android 11
iPhone 6 S Plus//iOS 14.4

**Comment by sergeysemenuta** [comment]
The issues are fixed. Works as expected on  the iOS and Android devices
Checked on:
Staging
ML v 0.16.19
iPhone X // iOS 13.3
Pixel 4XL//Android 11
